machine/machine_core/machine.py

```diff
@@ -68,7 +68,10 @@ def __init__(self, address="127.0.0.1", image="unknown", verbose=False, label=No
         if not browser:
             browser = address
 
-        super(Machine, self).__init__(user, ssh_address, ssh_port, identity_file, verbose=verbose)
+        if not label and image != "unknown":
+            label = "{}-{}-{}".format(image, ssh_address, ssh_port)
```

Comment:
I mean, once more around if you feel like turning that into an f-string, but otherwise I wouldn't worry...